File: problems/Python/121. Best Time to Buy and Sell Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minp = float('inf')
        maxp = 0
        
        for p in prices:
            if p < minp:
                minp = p
            elif p - minp > maxp:
                maxp = p - minp
                
        return maxp

File: problems/C++/121. Best Time to Buy and Sell Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minp = 2147483647;
        int maxp = 0;
        
        for (int p : prices) {
            if (p < minp) {
                minp = p;
            } else if (p - minp > maxp) {
                maxp = p - minp;
            }
        }
        return maxp;
    }
};